ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -372,24 +422,22 @@ std::tuple<ttnn::Shape, ttnn::MemoryConfig, bool> get_conv_padded_input_shape_an
         }
     }
     ParallelConfig parallel_config = input_tensor_parallel_config;
+    TensorMemoryLayout shard_layout = conv_config.shard_layout;
     if (conv_config.reshard_if_not_optimal || needs_shard_or_reshard) {
         auto block_shard_orientation =
             conv_config.transpose_shards ? ShardOrientation::COL_MAJOR : ShardOrientation::ROW_MAJOR;
+        if (shard_layout == TensorMemoryLayout::NONE) {
```

Comment:
one possibility ctould be to make `shard_layout` `std::optional`, then we won't need an explicit `NONE` layout.